File: 07-stacks-queues/05_next_greater.c
#include "../test_runner.h"

void next_greater_element(int* nums1, int n1, int* nums2, int n2, int* out) {
    for (int i = 0; i < n1; i++) out[i] = -1;
}

int main() {
    int out[10];
    next_greater_element(ARR(4,1,2), 3, ARR(1,3,4,2), 4, out);
    TEST_ARR("basic", out, ARR(-1,3,-1), 3);
    next_greater_element(ARR(2,4), 2, ARR(1,2,3,4), 4, out);
    TEST_ARR("simple", out, ARR(3,-1), 2);
    TEST_SUMMARY();
}

File: 07-stacks-queues/05_next_greater.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Next Greater Element I (LC #496) -- Easy
# For each element in nums1, find next greater element in nums2.
#
#   Key: Monotonic decreasing stack. Process nums2, map each element to its next greater.

def next_greater_element(nums1, nums2):
    pass


run_tests(next_greater_element, [
    (([4,1,2], [1,3,4,2]),  [-1,3,-1]),
    (([2,4], [1,2,3,4]),    [3,-1]),
])
